src/lib/addTags.ts

```
@@ -0,0 +1,50 @@
+  for (const node of body.childNodes) {
+    if (node.nodeType !== Node.TEXT_NODE) {
+      continue;
+    }
+
+    const nodes: Node[] = [];
+
+    const tagRegexp = /([^#]*)(#\w*)?/g;
+
+    let match: RegExpExecArray | null;
+
+    while ((match = tagRegexp.exec(node.textContent ?? ""))) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)